airbyte-integrations/connectors/source-commercetools/source_commercetools/source.py

```diff
@@ -0,0 +1,199 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from base64 import b64encode
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+# Basic full refresh stream
+class CommercetoolsStream(HttpStream, ABC):
+
+    # Page size
+    limit = 500
+    # Define primary key as sort key for full_refresh, or very first sync for incremental_refresh
+    primary_key = "id"
+    filter_field = "lastModifiedAt"
+    offset = 0
+
+    def __init__(self, region: str, project_key: str, start_date: str, client_id: str, client_secret: str, **kwargs):
+        super().__init__(**kwargs)
+        self.start_date = start_date
+        self.project_key = project_key
+        self.region = region
+        self.client_id = client_id
+        self.client_secret = client_secret
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        json_response = response.json()
+        total = json_response.get("total", 0)
+        offset = json_response.get("offset", 0)
```

Comment:
I know this is more secure to not break the connector but what do you think to use `total = json_response["total"]` if the object doesnt have total is better to break than enter a loop with 0. WDYT?

Reply:
Thanks for the review @marcosmarxm. I need to get the total if it exists or assign 0 to it because in using for a calculation on line 59. Im not use it in a for loop.